examples/demo-stf/README.md

```diff
@@ -33,90 +33,90 @@ several parameters that specify its exact behavior. In order, these four generic
 
-#### Implementing TxHooks
+```Rust
+impl<C: Context> TxHooks for Runtime<C> {
+    type Context = C;
 
-Once a transaction has passed stateless verification, it will get fed into the execution pipeline. In this pipeline there are four places
-where you can inject custom "hooks" using your `TxHooks` implementation.
+    fn pre_dispatch_tx_hook(
+        &self,
+        tx: Transaction<Self::Context>,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> anyhow::Result<<Self::Context as Spec>::Address> {
+        self.accounts.pre_dispatch_tx_hook(tx, working_set)
+    }
 
-1. At the beginning of the `apply_blob` function, before the blob is deserialized into a group of transactions. This is a good time to
-   apply per-batch validation logic like ensuring that the sequencer is properly bonded.
-2. Immediately before each transaction is dispatched to the runtime. This is a good time to apply stateful transaction verification, like checking
-   the nonce.
-3. Immediately after each transaction is executed. This is a good place to perform any post-execution operations, like incrementing the nonce.
-4. At the end of the `apply_blob` function. This is a good place to reward sequencers.
+    fn post_dispatch_tx_hook(
+        &self,
+        tx: &Transaction<Self::Context>,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> anyhow::Result<()> {
+        self.accounts.post_dispatch_tx_hook(tx, working_set)
+    }
+}
+```
 
-To use the `AppTemplate`, you need to provide a `TxHooks` implementation which specifies what needs to happen at each of these four
-stages.
+```Rust
+impl<C: Context> ApplyBlobHooks for Runtime<C> {
+    type Context = C;
 
-Its common for modules that need access to these hooks to export a `Hooks` struct. If you're relying on an unfamiliar module, be sure to check
-its documentation to make sure that you know about any hooks that it may rely on. Your `TxHooks` implementation will usually
-just be a wrapper which invokes each of these modules hooks. In this demo, we only rely
-on two modules which need access to the hooks - `sov-accounts` and `sequencer-registry`, so our `TxHooks` implementation only has two fields.
+    fn lock_sequencer_bond(
+        &self,
+        sequencer: &[u8],
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> anyhow::Result<()> {
+        self.sequencer.lock_sequencer_bond(sequencer, working_set)
+    }
 
-```rust
-pub struct DemoAppTxHooks<C: Context> {
-    accounts_hooks: accounts::hooks::Hooks<C>,
-    sequencer_hooks: sov_sequencer_registry::hooks::Hooks<C>,
+    fn reward_sequencer(
+        &self,
+        amount: u64,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> anyhow::Result<()> {
+        self.sequencer.reward_sequencer(amount, working_set)
+    }
 }
 ```
 
```

Comment:
nit: Can we make this section a sub-section of `Implementing Runtime`?

Reply:
Done